src/main/kotlin/io/github/gradlenexus/publishplugin/ReleaseNexusStagingRepository.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.gradlenexus.publishplugin
+
+import io.github.gradlenexus.publishplugin.internal.NexusClient
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+import org.gradle.kotlin.dsl.property
+import javax.inject.Inject
+
+@Suppress("UnstableApiUsage")
+open class ReleaseNexusStagingRepository @Inject
+constructor(objects: ObjectFactory, extension: NexusPublishExtension, repository: NexusRepository) :
+        BaseOperationOnNexusStagingRepository(objects, extension, repository) {
+
+    @get:Input
+    val stagingRepositoryId: Property<String> = objects.property()
+
+    init {
+        stagingRepositoryId.set(repository.stagingRepositoryId)
```

Comment:
Should this be `val stagingRepositoryId: Property<String> = objects.property().convention(repository.stagingRepositoryId)` ?

Reply:
It could be. Assuming someone will want to set the value in a task directly (instead of playing with the `repository` structure).

Reply:
Btw, `Property.convention()` was added in 5.1. Should we care about 4.10 users? Or having 6.0 on the doorstep it's not a problem?

Reply:
Let's stay compatible with 4.10 for now. That's the reason I didn't use `convention()` in the other places as well.